astropy/units/function/core.py

```diff
@@ -360,10 +360,12 @@ def to_string(self, format='generic'):
             The name of a format or a formatter object.  If not
             provided, defaults to the generic format.
         """
-        if format not in ('generic', 'unscaled', 'latex'):
+        if format not in ('generic', 'unscaled', 'latex', 'cds'):
             raise ValueError("Function units cannot be written in {} format. "
                              "Only 'generic', 'unscaled' and 'latex' are "
                              "supported.".format(format))
+        if format == 'cds':
```

Comment:
I think this is the right idea, but it should not be here, but rather in `DexUnit` (which is the only one that is allowed to be overridden; CDS cannot reproduce `mag` and `dB` function units). I.e., create a new method `to_string` there which has your special case for CDS, but otherwise just does `return super().to_string(....)`

Reply:
Inside UnitDex what should I do if the format is not cds?

Reply:
`return super().to_string(....)`